src/ToDoListView.test.js

```diff
@@ -137,4 +129,45 @@ describe('To Do List View', () => {
       expect(screen.getByText('Item 3')).toBeInTheDocument();
     });
   });
+  describe('Delete', () => {
+    it('delete button should send delete request', () => {
+      ToDoRequestService.deleteToDo.mockImplementation(() => new Promise(jest.fn()));
+      userEvent.click(itemOneDeleteButton());
+
+      expect(ToDoRequestService.deleteToDo).toBeCalledWith(1);
+    });
+
+    it('delete response should remove item from list', async () => {
+      ToDoRequestService.deleteToDo.mockResolvedValue({});
+
+      await act(async () => {
+        userEvent.click(itemOneDeleteButton());
+      });
+
+      expect(screen.queryByText('Item 1')).not.toBeInTheDocument();
+    });
+  });
+  describe('Edit', () => {
+    beforeEach(() => {
+      userEvent.click(editButton());
+    });
+    it('should open edit dialogue box when clicked', () => {
+      expect(screen.queryByText('Edit')).toBeInTheDocument();
+    });
+    it('should contain text box', () => {
+      expect(editTextBox()).toBeInTheDocument();
+    });
+    it('text box should contain To Do name', () => {
+      expect(editTextBox()).toHaveValue('Item 1');
+    });
+    it('dialog should contain save button', () => {
+      expect(screen.getByRole('button', { name: 'save-edited-to-do' }));
+    });
+    it('dialog should contain cancel button', () => {
+      expect(screen.getByRole('button', { name: 'cancel-edit-to-do' }));
```

Comment:
Why is the button name this strange kebab case, instead of just 'Cancel'?

Reply:
Does it need to be unique as a label? That was my thinking behind the verbose label